app/src/main/java/com/moyerun/moyeorun_android/views/BadgeRoundImageView.kt

```diff
@@ -0,0 +1,89 @@
+package com.moyerun.moyeorun_android.views
+import android.content.Context
+import android.content.res.TypedArray
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.moyerun.moyeorun_android.R
+import com.moyerun.moyeorun_android.databinding.ViewBadgeRoundImageBinding
+
+class BadgeRoundImageView : ConstraintLayout {
+    constructor(context: Context): super(context) { initView() }
+    constructor(context: Context, attrs: AttributeSet): super(context, attrs) {
+        initView()
+        setAttrs(attrs)
+    }
+    constructor(context: Context, attrs: AttributeSet, defStyle: Int): super(context, attrs) {
+        initView()
+        setAttrs(attrs, defStyle)
+    }
```

Comment:
`@JvmOverloads` 로 줄일 수 있을 것 같아요. AttributeSet 처리는 `init { }` 블럭에서 처리하구요
```suggestion
class BadgeRoundImageView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : ConstraintLayout(context, attrs, defStyleAttr) {
```

Reply:
사실 CustomVIew를 만들때 [Do not always trust @JvmOverloads](https://medium.com/@mmlodawski/https-medium-com-mmlodawski-do-not-always-trust-jvmoverloads-5251f1ad2cfe)에 대한 글도 있고 @JvmOverloads를 사용하지 않았는데 보일러플레이트 코드가 발생한다는 측면에서 조금 더 주의해서 사용하면 괜찮을 것 같아서 정상적으로 작동하는 것을 보고 변경해놓도록 하겠습니다!

Reply:
넵 말씀하신대로 `@JvmOverloads` 는 항상 기대한 결과를 보여줄 수 있는 기능은 아닙니다. 하지만 그건 defStyle가 null인 경우, 부모 클래스의 defStyle 을 따르지 않고 0으로 적용하기 때문에 발생하는 문제라서요, BadgeRoundImageView는 `ConstraltLayout`을 구현하고 있어서 스타일 이슈는 없어보입니다. 그래도 확인은 한 번 해보면 좋긴하겠네요 ㅎㅎ